nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,913 @@
+import io
+import string
+import os
+
+from collections import OrderedDict
+
+import numpy as np
+
+try:
+    from urllib.parse import quote
+except ImportError:
+    from urllib import quote  # Python2
+
+try:
+    from html import escape
+except ImportError:
+    from cgi import escape
+
+import pandas as pd
+import nibabel
+
+from matplotlib import pyplot as plt
+from nilearn.plotting import (plot_glass_brain,
+                              plot_roi,
+                              plot_stat_map,
+                              )
+from nilearn.plotting.img_plotting import MNI152TEMPLATE
+from nilearn.plotting.js_plotting_utils import HTMLDocument
+
+import nistats
+from nistats.reporting import (plot_contrast_matrix,
+                               plot_design_matrix,
+                               get_clusters_table,
+                               )
+from nistats.thresholding import map_threshold
+
+html_template_root_path = os.path.join(os.path.dirname(__file__),
+                                       'glm_reporter_templates')
+
+
+def make_glm_report(
+        model,
+        contrasts,
+        title='auto',
+        roi_img=None,
+        bg_img=MNI152TEMPLATE,
+        threshold=3.09,
+        alpha=0.01,
+        cluster_threshold=0,
+        height_control='fpr',
+        min_distance=8.,
+        plot_type='slice',
+        display_mode=None,
+        nb_width=1600,
+        nb_height=800,
+        ):
+    """ Returns HTMLDocument object for a report which shows all important aspects of a fitted GLM. The object can be opened in a browser, displayed in a notebook, or saved to disk as a portable HTML file.
+    
+    Usage:
+        report = make_glm_report(model, contrasts)
+        report.open_in_browser()
+        report.save_as_html(destination_path)
+    
+    Parameters
+    ----------
+    model: FirstLevelModel or SecondLevelModel object
+        A fitted first or second level model object.
+        Must have the computed design matrix(ces).
+        
+    contrasts: Dict[string, ndarray] , String, List[String], ndarray, List[ndarray]
+        Contrasts information for a first or second level model.
+        Corresponds to the ``contrast_def`` for the FirstLevelModel [1]_
+        & second_level_contrast for a SecondLevelModel [2]_ .
+    
+    title: String or 3 element List/Tuple, default 'auto'
+        If string, represents the web page's title and primary heading,
+        model type is sub-heading.
+        If 3 element List/Tuple, represents page title heading, sub-heading.
+        If 'auto', uses the contrast titles to generate a title.
+        
+    roi_img : Niimg-like object
```

Comment:
Indeed I've just realized that it is actually bad to introduce another roi than the one that `glm.mask_img_` since we are reporting on THIS glm with THIS mask image.